pkg/api/pluginproxy/access_token_provider_test.go

```diff
@@ -91,4 +99,217 @@ func TestAccessToken(t *testing.T) {
 			So(token2, ShouldEqual, "abc")
 		})
 	})
+
+	Convey("Plugin with token auth route", t, func() {
+		apiHandler := http.NewServeMux()
+		server := httptest.NewServer(apiHandler)
+
+		defer server.Close()
+		pluginRoute := &plugins.AppPluginRoute{
+			Path:   "pathwithtokenauth1",
+			Url:    "",
+			Method: "GET",
+			TokenAuth: &plugins.JwtTokenAuth{
+				Url: server.URL + "/oauth/token",
+				Scopes: []string{
+					"https://www.testapi.com/auth/monitoring.read",
+					"https://www.testapi.com/auth/cloudplatformprojects.readonly",
+				},
+				Params: map[string]string{
+					"grant_type":    "client_credentials",
+					"client_id":     "{{.JsonData.client_id}}",
+					"client_secret": "{{.SecureJsonData.client_secret}}",
+					"audience":      "{{.JsonData.audience}}",
+					"client_name":   "datasource_plugin",
+				},
+			},
+		}
+
+		templateData := templateData{
+			JsonData: map[string]interface{}{
+				"client_id": "my_client_id",
+				"audience":  "www.example.com",
+			},
+			SecureJsonData: map[string]string{
+				"client_secret": "my_secret",
+			},
+		}
+
+		ds := &models.DataSource{Id: 1, Version: 2}
+
+		var i int
```

Comment:
Can we delete this one?

Reply:
used for the creation of the AccessTokenProvider, but we could create a empty DataSource instead.

Reply:
`&models.DataSource{}` can be used as a argument for  `newAccessTokenProvider`

Reply:
but the variable `i` is not used.

Reply:
Oh, i thought it was `ds`. Anyways, It will be used in the next push.